xds/googledirectpath/googlec2p_test.go

```diff
@@ -251,3 +253,21 @@ func TestBuildXDS(t *testing.T) {
 		})
 	}
 }
+
+// TestDialFailsWhenTargetContainsAuthority attempts to Dial a target URI of
+// google-c2p scheme with a non-empty authority and verifies that it fails with
+// an expected error.
+func TestBuildFailsWhenCalledWithAuthority(t *testing.T) {
+	uri := "google-c2p://an-authority/resource"
+	cc, err := grpc.Dial(uri, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	defer func() {
+		if cc != nil {
+			cc.Close()
+		}
+	}()
+	wantErr := "google-c2p URI scheme does not support authorities"
+	if err == nil || !strings.Contains(err.Error(), wantErr) {
+		cc.Close()
```

Comment:
Just realized that we don't need this `cc.Close()` here since that will be taken care of by the deferred function.

Reply:
updating that and merging